talk/morelanguage/move.tex

```diff
@@ -205,24 +208,47 @@
     \begin{cppcode*}{}
       class Movable {
         Movable();
-        Movable(Movable &&other) :
+        Movable(const Movable &other);
+        Movable(Movable &&other) noexcept :
           Movable() {         // constructor delegation
           swap(*this, other);
         }
-        Movable& operator=(Movable other) { // by value
+        Movable& operator=(Movable other) noexcept { // by value
           swap(*this, other);
           return *this;
         }
-        friend void swap(Movable &a, Movable &b);
+        friend void swap(Movable &a, Movable &b) noexcept {...}
       };
-      void swap(Movable &a, Movable &b);
       Movable a, b;
       a = b;            // operator= copies b into "other"
       a = std::move(b); // operator= moves b into "other"
     \end{cppcode*}
   \end{exampleblock}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitlecpp[11]{Move semantics : the complete picture}
+  \begin{exampleblock}{Practically}
+    \small
+    \begin{cppcode*}{}
+      class Movable {
+        Movable();
+        Movable(const Movable &other);
+        Movable(Movable &&other) noexcept :
+          Movable() {         // constructor delegation
+          swap(*this, other);
+        }
+        Movable& operator=(const Movable& other);
+        Movable& operator=(Movable&& other) noexcept {
+          swap(*this, other);
+          return *this;
+        }
+        friend void swap(Movable &a, Movable &b) noexcept { ... }
+      };
+    \end{cppcode*}
+  \end{exampleblock}
+\end{frame}
+
```

Comment:
I also felt we should show the version without the unified copy/move assignment operator.

Reply:
I found it hard to spot the differences, as the code moves up and down. I suggest to to beamer's `\begin{overprint}` inside the green box, so the box would stay stationary, and the code would start at the identical location.

Does this make sense?

Reply:
I tried it but then there was more vertical space between the frame titel and the exampleblock and the code did no longer fit. I now aligned the frame content at the top, that works better.